lib/spack/spack/cmd/env.py

```diff
@@ -170,12 +170,21 @@ def env_create_setup_parser(subparser):
         help='specify that this environment should maintain a view at the'
              ' specified path (by default the view is maintained in the'
              ' environment directory)')
+    view_opts.add_argument(
+        '--copy',
+        help='when adding a view, use copy instead of link.')
```

Comment:
Views are configured via a section in the environment yaml -- see [this part of the spack.yaml schema](https://github.com/spack/spack/blob/develop/lib/spack/spack/schema/env.py#L111).

```yaml
spack:
    view:
        root: <string>             # path to the root of the view
        link: roots|all            # roots means only put roots in the view, all includes roots and all deps
        select: [list of specs]    # fine-tune what gets included (list of spec matchers)
        select: [list of specs]    # fine-tune what gets included (list of spec matchers)
        projections:
            <spec>: <path-format>     # customize where things are linked (e.g., for combinatorial views)
```

We really need some new config option here so you can customize how the view gets created.  How about:

```yaml
spack:
    view:
        link_type: symlink|hardlink|copy
```

I can't think of a better name than `link_type` -- suggestions welcome.  It's a bit overloaded with our existing `link:` option.

Reply:
I couldn't get the basic (command line) functionality working (see error above) but if I could, I could definitely add that spack.yaml update too.